activesupport/lib/active_support/current_attributes.rb

```diff
@@ -137,6 +137,7 @@ def attribute(*names, default: NOT_SET)
         Delegation.generate(singleton_class, names, to: :instance, nilable: false, signature: "")
         Delegation.generate(singleton_class, names.map { |n| "#{n}=" }, to: :instance, nilable: false, signature: "value")
 
+        current_instances.delete(current_instances_key)
```

Comment:
I think it would be better if we avoided the call to `instance`, we can probably do it by not using `respond_to?` but `singleton_class.method_defined?`.